src/tarray.jl

```diff
@@ -96,52 +145,47 @@ Array(tz)                     # convert to 4-element Array{Float64,1}:  [9.0  9.
 ```
 """
 function tfill(val::Real, dim)
-    res = TArray{typeof(val),length(dim)}();
+    res = TArray{typeof(val), length(dim), Array{typeof(val), length(dim)}}();
     n = n_copies()
     d = fill(val,dim)
-    task_local_storage(res.ref, (n,d))
+    _local_storage(res, (n, d))
     return res
 end
 
 #
 # Conversion between TArray and Array
 #
-_get(x) = x
-function _get(x::TArray)
-    n, d = task_local_storage(x.ref)
-    return d
-end
 
 function Base.convert(::Type{Array}, x::TArray)
     return convert(Array{eltype(x), ndims(x)}, x)
 end
-function Base.convert(::Type{Array{T,N}}, x::TArray{T,N}) where {T,N}
+function Base.convert(::Type{Array{T, N}}, x::TArray{T, N}) where {T, N}
     c = convert(Array{T, N}, deepcopy(_get(x)))
     return c
 end
 
 function Base.convert(::Type{TArray}, x::AbstractArray)
-    return convert(TArray{eltype(x),ndims(x)}, x)
+    return convert(TArray{eltype(x), ndims(x)}, x)
 end
-function Base.convert(::Type{TArray{T,N}}, x::AbstractArray{T,N}) where {T,N}
-    res = TArray{T,N}()
+function Base.convert(::Type{TArray{T, N}}, x::AbstractArray{T, N}) where {T, N}
+    res = TArray{T, N, typeof(x)}()
     n   = n_copies()
-    task_local_storage(res.ref, (n,x))
+    _local_storage(res, (n, x))
     return res
 end
 
 #
 # Representation
 #
 function Base.show(io::IO, ::MIME"text/plain", x::TArray)
-    arr = x.orig_task.storage[x.ref][2]
+    arr = x.data[x.orig_task][2]
     @warn "Here shows the originating task's storage, " *
         "not the current task's storage. " *
         "Please explicitly call show(::TArray) to display the current task's version of a TArray."
     show(io,  MIME("text/plain"), arr)
 end
 
-Base.show(io::IO, x::TArray) = Base.show(io::IO, task_local_storage(x.ref)[2])
+Base.show(io::IO, x::TArray) = show(io::IO, _get(x))
```

Comment:
Just a minor suggestion:
```suggestion
Base.show(io::IO, x::TArray) = show(io, _get(x))
```